src/guards/type-guards-alt.ts

```diff
@@ -0,0 +1,6 @@
+import { GraphQLScalarType } from "graphql";
+
+// no wrapping class. function returns a plain ol boolean
+export function isScalar(type: unknown): boolean {
```

Comment:
this generally works, but for more safety, type predicates are the way to go

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
yes, but my understanding is that `instanceof` is itself a type predicate and you're just wrapping it in _another_ one. 😄